yt/frontends/cholla/fields.py

```diff
@@ -0,0 +1,155 @@
+import numpy as np
+from unyt import Zsun
+
+from yt.fields.field_info_container import FieldInfoContainer
+from yt.utilities.physical_constants import kboltz, mh
+
+# Copied from Athena frontend
+pres_units = "code_pressure"
+erg_units = "code_mass * (code_length/code_time)**2"
+rho_units = "code_mass / code_length**3"
+mom_units = "code_mass / code_length**2 / code_time"
+
+
+def velocity_field(comp):
+    def _velocity(field, data):
+        return data["cholla", f"momentum_{comp}"] / data["cholla", "density"]
+
+    return _velocity
+
+
+class ChollaFieldInfo(FieldInfoContainer):
+    known_other_fields = (
+        # Each entry here is of the form
+        # ( "name", ("units", ["fields", "to", "alias"], # "display_name")),
+        ("density", (rho_units, ["density"], None)),
+        ("momentum_x", (mom_units, ["momentum_x"], None)),
+        ("momentum_y", (mom_units, ["momentum_y"], None)),
+        ("momentum_z", (mom_units, ["momentum_z"], None)),
+        ("Energy", ("code_pressure", ["total_energy_density"], None)),
+        ("scalar0", (rho_units, [], None)),
+        ("metal_density", (rho_units, ["metal_density"], None)),
+    )
+
+    known_particle_fields = ()
+
+    def __init__(self, ds, field_list):
+        super().__init__(ds, field_list)
```

Comment:
I don't see the point of adding a frame to the call stack here
```suggestion
```

Reply:
I found it in the skeleton, and it seems to make sense?  Aren't we just inheriting any sort of field definitions from the superclass?

Reply:
yes we are, but here we override the base `__init__` method just to call it again one stack level deeper, so it accomplishes exactly nothing more than if we just inherited the base method.
I don't actually see a good reason for having this in the skeleton frontend.